registry/remote/repository.go

```diff
@@ -126,6 +129,21 @@ func NewRepository(reference string) (*Repository, error) {
 	}, nil
 }
 
+func newRepositoryWithOptions(ref registry.Reference, opts *RepositoryOptions) (*Repository, error) {
+	if err := ref.ValidateRepository(); err != nil {
+		return nil, err
+	}
+	return &Repository{
+		Client:               opts.Client,
+		Reference:            ref,
+		PlainHTTP:            opts.PlainHTTP,
+		ManifestMediaTypes:   slices.Clone(opts.ManifestMediaTypes),
```

Comment:
Consider the following code to remove the dependency on `golang.org/x/exp`.
```suggestion
		ManifestMediaTypes:   append([]string{}, opts.ManifestMediaTypes...),
```
Of course, we can have a `if` check if `nil` matters.